lib/ts/recipe/passwordless/recipeImplementation.ts

```diff
@@ -0,0 +1,275 @@
+                        ...context,
+                        action: "PASSWORDLESS_CONSUME_CODE",
+                    });
+                }
+            );
+
+            if (response.status === "RESTART_FLOW_ERROR") {
+                input.config.onHandleEvent({
+                    action: "PASSWORDLESS_RESTART_FLOW",
+                });
+            } else if (response.status === "OK") {
+                input.config.onHandleEvent({
+                    action: "SUCCESS",
+                    isNewUser: response.createdUser,
+                    user: response.user,
+                });
+            }
+            return response;
+        },
+        doesEmailExist: async function (input: { email: string; config: NormalisedConfig }): Promise<boolean> {
+            const response: ExistsAPIResponse = await querier.get(
+                "/signup/email/exists",
+                {},
+                { email: input.email },
+                (context) => {
+                    return input.config.preAPIHook({
+                        ...context,
+                        action: "EMAIL_EXISTS",
+                    });
+                }
+            );
+
+            return response.exists;
+        },
+
+        doesPhoneNumberExist: async function (input: {
+            phoneNumber: string;
+            config: NormalisedConfig;
+        }): Promise<boolean> {
+            const response: ExistsAPIResponse = await querier.get(
+                "/signup/phoneNumber/exists",
+                {},
+                { phoneNumber: input.phoneNumber },
+                (context) => {
+                    return input.config.preAPIHook({
+                        ...context,
+                        action: "PHONE_NUMBER_EXISTS",
+                    });
+                }
+            );
+
+            return response.exists;
+        },
+        getLoginAttemptInfo: function ():
+            | undefined
+            | {
+                  deviceId: string;
+                  preAuthSessionId: string;
+                  flowType: FlowType;
+                  contactInfo: string;
+                  contactMethod: "EMAIL" | "PHONE";
+                  lastResend: number;
+              } {
+            const storedInfo = getLocalStorage(PASSWORDLESS_LOGIN_ATTEMPT_INFO_STORAGE_KEY);
+            if (!storedInfo) {
+                return undefined;
+            }
+            try {
+                const info = JSON.parse(storedInfo);
+
+                return {
+                    contactInfo: info.contactInfo,
+                    contactMethod: info.contactMethod,
+                    deviceId: info.deviceId,
+                    flowType: info.flowType,
+                    preAuthSessionId: info.preAuthSessionId,
+                    lastResend: info.lastResend,
+                };
+            } catch (ex) {
+                return undefined;
+            }
+        },
+        setLoginAttemptInfo: function (input: {
```

Comment:
You also need to store redirectToPath (optionally) in here? So that custom redirection via magic link flow works too.